File: posixlib/src/main/resources/scala-native/langinfo.c
#if defined(SCALANATIVE_COMPILE_ALWAYS) || defined(__SCALANATIVE_POSIX_LANGINFO)
#if defined(__unix__) || defined(__unix) || defined(unix) ||                   \
    (defined(__APPLE__) && defined(__MACH__))

#include <langinfo.h>

#if defined(__OpenBSD__)
#define ERA -1
#define ERA_D_FMT -1
#define ERA_D_T_FMT -1
#define ERA_T_FMT -1
#define ALT_DIGITS -1
#endif // OpenBSD

int scalanative_codeset() { return CODESET; };

int scalanative_d_t_fmt() { return D_T_FMT; };

int scalanative_d_fmt() { return D_FMT; };

int scalanative_t_fmt() { return T_FMT; };

int scalanative_t_fmt_ampm() { return T_FMT_AMPM; };

int scalanative_am_str() { return AM_STR; };

int scalanative_pm_str() { return PM_STR; };

int scalanative_day_1() { return DAY_1; };

int scalanative_day_2() { return DAY_2; };

int scalanative_day_3() { return DAY_3; };

int scalanative_day_4() { return DAY_4; };

int scalanative_day_5() { return DAY_5; };

int scalanative_day_6() { return DAY_6; };

int scalanative_day_7() { return DAY_7; };

int scalanative_abday_1() { return ABDAY_1; };

int scalanative_abday_2() { return ABDAY_2; };

int scalanative_abday_3() { return ABDAY_3; };

int scalanative_abday_4() { return ABDAY_4; };

int scalanative_abday_5() { return ABDAY_5; };

int scalanative_abday_6() { return ABDAY_6; };

int scalanative_abday_7() { return ABDAY_7; };

int scalanative_mon_1() { return MON_1; };

int scalanative_mon_2() { return MON_2; };

int scalanative_mon_3() { return MON_3; };

int scalanative_mon_4() { return MON_4; };

int scalanative_mon_5() { return MON_5; };

int scalanative_mon_6() { return MON_6; };

int scalanative_mon_7() { return MON_7; };

int scalanative_mon_8() { return MON_8; };

int scalanative_mon_9() { return MON_9; };

int scalanative_mon_10() { return MON_10; };

int scalanative_mon_11() { return MON_11; };

int scalanative_mon_12() { return MON_12; };

int scalanative_abmon_1() { return ABMON_1; };

int scalanative_abmon_2() { return ABMON_2; };

int scalanative_abmon_3() { return ABMON_3; };

int scalanative_abmon_4() { return ABMON_4; };

int scalanative_abmon_5() { return ABMON_5; };

int scalanative_abmon_6() { return ABMON_6; };

int scalanative_abmon_7() { return ABMON_7; };

int scalanative_abmon_8() { return ABMON_8; };

int scalanative_abmon_9() { return ABMON_9; };

int scalanative_abmon_10() { return ABMON_10; };

int scalanative_abmon_11() { return ABMON_11; };

int scalanative_abmon_12() { return ABMON_12; };

int scalanative_era() { return ERA; };

int scalanative_era_d_fmt() { return ERA_D_FMT; };

int scalanative_era_d_t_fmt() { return ERA_D_T_FMT; };

int scalanative_era_t_fmt() { return ERA_T_FMT; };

int scalanative_alt_digits() { return ALT_DIGITS; };

int scalanative_radixchar() { return RADIXCHAR; };

int scalanative_thousep() { return THOUSEP; };

int scalanative_yesexpr() { return YESEXPR; };

int scalanative_noexpr() { return NOEXPR; };

int scalanative_crncystr() { return CRNCYSTR; };

#endif // Unix or Mac OS
#endif

File: posixlib/src/main/scala/scala/scalanative/posix/langinfo.scala
package scala.scalanative
package posix

import scala.scalanative.unsafe._

/** POSIX langinfo.h for Scala
 *
 *  The Open Group Base Specifications
 *  [[https://pubs.opengroup.org/onlinepubs/9699919799 Issue 7, 2018]] edition.
 */

@extern object langinfo {

  type locale_t = locale.locale_t

  type nl_item = nl_types.nl_item

// Symbolic constants

  @name("scalanative_codeset")
  def CODESET: CInt = extern

  @name("scalanative_d_t_fmt")
  def D_T_FMT: CInt = extern

  @name("scalanative_d_fmt")
  def D_FMT: CInt = extern

  @name("scalanative_t_fmt")
  def T_FMT: CInt = extern

  @name("scalanative_t_fmt_ampm")
  def T_FMT_AMPM: CInt = extern

  @name("scalanative_am_str")
  def AM_STR: CInt = extern

  @name("scalanative_pm_str")
  def PM_STR: CInt = extern

  @name("scalanative_day_1")
  def DAY_1: CInt = extern

  @name("scalanative_day_2")
  def DAY_2: CInt = extern

  @name("scalanative_day_3")
  def DAY_3: CInt = extern

  @name("scalanative_day_4")
  def DAY_4: CInt = extern

  @name("scalanative_day_5")
  def DAY_5: CInt = extern

  @name("scalanative_day_6")
  def DAY_6: CInt = extern

  @name("scalanative_day_7")
  def DAY_7: CInt = extern

  @name("scalanative_abday_1")
  def ABDAY_1: CInt = extern

  @name("scalanative_abday_2")
  def ABDAY_2: CInt = extern

  @name("scalanative_abday_3")
  def ABDAY_3: CInt = extern

  @name("scalanative_abday_4")
  def ABDAY_4: CInt = extern

  @name("scalanative_abday_5")
  def ABDAY_5: CInt = extern

  @name("scalanative_abday_6")
  def ABDAY_6: CInt = extern

  @name("scalanative_abday_7")
  def ABDAY_7: CInt = extern

  @name("scalanative_mon_1")
  def MON_1: CInt = extern

  @name("scalanative_mon_2")
  def MON_2: CInt = extern

  @name("scalanative_mon_3")
  def MON_3: CInt = extern

  @name("scalanative_mon_4")
  def MON_4: CInt = extern

  @name("scalanative_mon_5")
  def MON_5: CInt = extern

  @name("scalanative_mon_6")
  def MON_6: CInt = extern

  @name("scalanative_mon_7")
  def MON_7: CInt = extern

  @name("scalanative_mon_8")
  def MON_8: CInt = extern

  @name("scalanative_mon_9")
  def MON_9: CInt = extern

  @name("scalanative_mon_10")
  def MON_10: CInt = extern

  @name("scalanative_mon_11")
  def MON_11: CInt = extern

  @name("scalanative_mon_12")
  def MON_12: CInt = extern

  @name("scalanative_abmon_1")
  def ABMON_1: CInt = extern

  @name("scalanative_abmon_2")
  def ABMON_2: CInt = extern

  @name("scalanative_abmon_3")
  def ABMON_3: CInt = extern

  @name("scalanative_abmon_4")
  def ABMON_4: CInt = extern

  @name("scalanative_abmon_5")
  def ABMON_5: CInt = extern

  @name("scalanative_abmon_6")
  def ABMON_6: CInt = extern

  @name("scalanative_abmon_7")
  def ABMON_7: CInt = extern

  @name("scalanative_abmon_8")
  def ABMON_8: CInt = extern

  @name("scalanative_abmon_9")
  def ABMON_9: CInt = extern

  @name("scalanative_abmon_10")
  def ABMON_10: CInt = extern

  @name("scalanative_abmon_11")
  def ABMON_11: CInt = extern

  @name("scalanative_abmon_12")
  def ABMON_12: CInt = extern

  @name("scalanative_era")
  def ERA: CInt = extern

  @name("scalanative_era_d_fmt")
  def ERA_D_FMT: CInt = extern

  @name("scalanative_era_d_t_fmt")
  def ERA_D_T_FMT: CInt = extern

  @name("scalanative_era_t_fmt")
  def ERA_T_FMT: CInt = extern

  @name("scalanative_alt_digits")
  def ALT_DIGITS: CInt = extern

  @name("scalanative_radixchar")
  def RADIXCHAR: CInt = extern

  @name("scalanative_thousep")
  def THOUSEP: CInt = extern

  @name("scalanative_yesexpr")
  def YESEXPR: CInt = extern

  @name("scalanative_noexpr")
  def NOEXPR: CInt = extern

  @name("scalanative_crncystr")
  def CRNCYSTR: CInt = extern

// Methods

  def nl_langinfo(item: nl_item): CString = extern

  def nl_langinfo_l(item: nl_item, locale: locale_t): CString = extern
}
